src/api/emailNotifications/client.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpError } from 'app/../error/AppErrors';
+
+import config from 'app/config';
+
+import createHttpClient from '../client';
+import {
+  SubscribePayload,
+  UnsubscribePayload,
+  AuthenticatePayload,
+  ModifyPayload,
+  CheckSubscriptionPayload,
+  VerifyEmailPayload,
+  VerifyEmailResponse,
+  SubscribeResponse,
+  ModifyResponse,
+  CheckSubscriptionResponse,
+} from './types';
+
+export enum EmailNotificationsError {
+  INVALID_EMAIL = 'Enter a valid email address',
+  ERROR_VERIFICATION_ERROR = 'On /verify-email/ request the server responded with result: "error"',
+  WRONG_PIN = 'Wrong pin',
+  ERROR_REQUEST_AUTHENTICATION = 'On /authenticate/ request the server responded with result: "error',
+}
+
+const httpClient = createHttpClient(config.emailNotificationsApi);
+
+export const subscribeEmail = async (data: SubscribePayload): Promise<SubscribeResponse> => {
+  const response = await httpClient.post(`/subscribe/`, data);
+
+  return response.data;
+};
+
+export const unsubscribeEmail = async (data: UnsubscribePayload): Promise<UnsubscribePayload> => {
+  const response = await httpClient.post(`/unsubscribe/`, data);
+
+  return response.data;
+};
+
+export const authenticate = async (data: AuthenticatePayload): Promise<AuthenticatePayload> => {
+  const response = await httpClient.post(`/authenticate/`, data);
+
+  if (response.data.result === 'error') {
```

Comment:
there is something wrong with the backend if it returns status 200 and an error, on other requests it throws an error

Reply:
Well, yea, agreed, but as Forrest Gump said: shit happens. That's why in general I'm against generic handlers - when an inconsistency like this occurs we must be ready to handle it.

Reply:
Do you recommend moving it back to Saga?

Reply:
as discussed with the backend team, we won't have the situation that there will be http 200 code and result = 'error', so this code can be removed

Reply:
Done